app/scripts/controllers/deployment.js

```diff
@@ -185,10 +189,14 @@ angular.module('openshiftConsole')
               group: 'extensions',
               resource: 'replicasets'
             }, context, function(replicaSetData) {
+              $scope.emptyMessage = "No deployments to show";
+
               var replicaSets = replicaSetData.by('metadata.name');
               replicaSets = OwnerReferencesService.filterForController(replicaSets, deployment);
               $scope.inProgressDeployment = _.chain(replicaSets).filter('status.replicas').size() > 1;
-              $scope.replicaSetsForDeployment = DeploymentsService.sortByRevision(replicaSets);
+
+              $scope.unfilteredReplicaSetsForDeployment = DeploymentsService.sortByRevision(replicaSets);
+              $scope.replicaSetsForDeployment = LabelFilter.getLabelSelector().select($scope.unfilteredReplicaSetsForDeployment);
```

Comment:
When we fix keeping filters on page reload, I think we need a call to `updateFilterWarning` here?